src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -150,92 +150,98 @@ class JesBackend extends Backend with LazyLogging {
     )
   }
 
-  override def setupCallEnvironment(call: Call, workflowDescriptor: WorkflowDescriptor): JesTaskExecutionContext = {
-    val callGcsPath = s"${workflowDescriptor.callDir(call)}"
-    JesTaskExecutionContext(GoogleCloudStoragePath(callGcsPath), JesConnection)
+  override def bindCall(workflowDescriptor: WorkflowDescriptor,
+                        call: Call,
+                        locallyQualifiedInputs: CallInputs,
+                        abortRegistrationFunction: AbortFunctionRegistration): BackendCall = {
+    JesBackendCall(this, workflowDescriptor, call, locallyQualifiedInputs, abortRegistrationFunction)
   }
 
-  override def executeCommand(instantiatedCommandLine: String,
-                              workflowDescriptor: WorkflowDescriptor,
-                              call: Call,
-                              backendInputs: CallInputs,
-                              scopedLookupFunction: ScopedLookupFunction,
-                              abortFunctionRegistration: AbortFunctionRegistration): Try[Map[String, WdlValue]] = {
+  override def execute(backendCall: BackendCall): Try[CallOutputs] = backendCall match {
+    case bc: JesBackendCall => execute(bc)
```

Comment:
I think the thing to do here is for BackendCall to know its implementing type (perhaps via an F-bound but but I think a simple abstract type would suffice). That way the `execute` methods all would only accept the appropriately typed BackendCall